### ISSUE TITLE
parse_expr(u'custom(x+x) + 1' , evaluate=False) crashes sympy

### ISSUE THREAD

**Issue by MrDarkHorse** [sympy/sympy#11443]
`parse_expr(u'custom(x+x)+1' , evaluate=False)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Python/2.7/site-packages/sympy/parsing/sympy_parser.py", line 892, in parse_expr
    code = compile(evaluateFalse(code), '<string>', 'eval')
  File "/Library/Python/2.7/site-packages/sympy/parsing/sympy_parser.py", line 902, in evaluateFalse
    node = EvaluateFalseTransformer().visit(node)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/ast.py", line 241, in visit
    return visitor(node)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/ast.py", line 297, in generic_visit
    value = self.visit(value)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/ast.py", line 241, in visit
    return visitor(node)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/ast.py", line 306, in generic_visit
    new_node = self.visit(old_value)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/ast.py", line 241, in visit
    return visitor(node)
  File "/Library/Python/2.7/site-packages/sympy/parsing/sympy_parser.py", line 956, in visit_BinOp
    new_node.args = self.flatten(new_node.args, sympy_class)
  File "/Library/Python/2.7/site-packages/sympy/parsing/sympy_parser.py", line 924, in flatten
    if isinstance(arg, ast.Call) and arg.func.id == func:
AttributeError: 'Call' object has no attribute 'id'`

This is also true if I insert the custom function into the global_dict
`global_dict['custom'] = custom_func`
`parse_expr(u'custom(x+x)+1' , evaluate=False, global_dict=global_dict)`

FWIW, these work just fine:
`parse_expr(u'custom(x+x)' , evaluate=False)`
`parse_expr(u'custom(x+x)' , evaluate=False, global_dict=global_dict)`

**Comment by oscarbenjamin** [comment]
Still the case now:
```julia
In [92]: from sympy.parsing.sympy_parser import parse_expr

In [93]: parse_expr(u'custom(x+x)+1' , evaluate=False)
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
...
AttributeError: 'Call' object has no attribute 'id'
```

**Comment by jksuom** [comment]
I get the same error with an ascii string.
```
In [6]: parse_expr('custom(x+x)+1' , evaluate=False)
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
...
/home/jks/sympy/sympy/parsing/sympy_parser.pyc in flatten(self, args, func)
    994         result = []
    995         for arg in args:
--> 996             if isinstance(arg, ast.Call) and arg.func.id == func:
    997                 result.extend(self.flatten(arg.args, func))
    998             else:

AttributeError: 'Call' object has no attribute 'id'
```
It seems that only 'Name' objects have attribute 'id'. That should be checked:
```
--- a/sympy/parsing/sympy_parser.py
+++ b/sympy/parsing/sympy_parser.py
@@ -993,7 +993,8 @@ class EvaluateFalseTransformer(ast.NodeTransformer):
     def flatten(self, args, func):
         result = []
         for arg in args:
-            if isinstance(arg, ast.Call) and arg.func.id == func:
+            if (isinstance(arg, ast.Call) and isinstance(arg.func, ast.Name)
+                and arg.func.id == func):
                 result.extend(self.flatten(arg.args, func))
             else:
                 result.append(arg)
```

**Comment by oscarbenjamin** [comment]
This was fixed in SymPy 1.5. Bisected the fix to f7750a57ad7bf09ad4629d5e6f10542e108039e3 from #17816
